src/main/java/it/pagopa/transactions/commands/handlers/TransactionSendClosureHandler.java

```diff
@@ -193,56 +172,22 @@ public Mono<Tuple2<Optional<TransactionRefundRequestedEvent>, Either<Transaction
-                                                                                                .getPspId()
-                                                                                )
-                                                                                .idChannel(
-                                                                                        transactionAuthorizationRequestData
-                                                                                                .getPspChannelCode()
-                                                                                )
-                                                                                .businessName(
-                                                                                        transactionAuthorizationRequestData
-                                                                                                .getPspBusinessName()
-                                                                                )
-                                                                )
-                                                )
-                                                .info(
-                                                        new InfoDto()
-                                                                .type(
-                                                                        transactionAuthorizationRequestData
-                                                                                .getPaymentTypeCode()
-                                                                )
-                                                                .brandLogo(
-                                                                        transactionAuthorizationRequestData.getLogo()
-                                                                                .toString()
-                                                                )
-                                                )
-                                                .user(new UserDto().type(UserDto.TypeEnum.GUEST))
-
                                 );
                     }
+                    ClosePaymentRequestV2Dto.OutcomeEnum outcomeEnum = closePaymentRequest.getOutcome();
+                    closePaymentRequest.transactionDetails(
+                            buildTransactionDetailsDto(
+                                    transactionActivatedData,
+                                    authRequestData,
+                                    transactionAuthorizationRequestData,
+                                    transactionAuthorizationCompletedData,
+                                    command,
+                                    fee,
+                                    amount,
+                                    totalAmount,
+                                    outcomeEnum
+                            )
+                    );
```

Comment:
This doesn't modify the original object but creates a new one. Is it necessary to split the method call?